bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveMultiLevelSensorConverter.java

```diff
@@ -106,16 +110,13 @@ public void handleEvent(ZWaveCommandClassValueEvent event, Item item, Map<String
 				switch(senType) {
 				case TEMPERATURE:
 					// For temperature, there are only two scales, so we simplify the conversion
-					if(sensorEvent.getSensorScale() == 0) {
-						// Scale is celsius, convert to fahrenheit
-						double c = ((BigDecimal)val).doubleValue();
-						val = new BigDecimal((c * 9.0 / 5.0) + 32.0 );
-					}
-					else if(sensorEvent.getSensorScale() == 1) {
-						// Scale is fahrenheit, convert to celsius
-						double f = ((BigDecimal)val).doubleValue();
-						val = new BigDecimal((f - 32.0) * 5.0 / 9.0 );					
-					}
+                    if (sensorEvent.getSensorScale() == 0) {
+                        // Scale is celsius, convert to fahrenheit
+                        val = val.multiply(ONE_POINT_EIGHT).add(THIRTY_TWO).movePointRight(1);
```

Comment:
The movePointRight and Left were for a vendor-specific F*10 to/from C implementation, so drop those! :)